plugin/backend_linux.go

```diff
@@ -294,11 +294,11 @@ func (pm *Manager) Remove(name string, config *types.PluginRmConfig) error {
 	}
 
 	id := p.GetID()
-	pm.pluginStore.Remove(p)
 	pluginDir := filepath.Join(pm.libRoot, id)
 	if err := os.RemoveAll(pluginDir); err != nil {
-		logrus.Warnf("unable to remove %q from plugin remove: %v", pluginDir, err)
+		return errors.Wrap(err, "failed to remove plugin directory")
```

Comment:
I think if someone is specifying force remove then we should remove it from the store no matter what.
We could report a warning still.

This is what we do for containers and volumes.